src/main/java/com/ibm/crypto/plus/provider/RSAKeyFactory.java

```diff
@@ -124,6 +128,59 @@ static void checkKeyLengths(int modulusLen, BigInteger exponent, int minModulusL
         }
     }
 
+    /**
+     * Check the length of an RSA key modulus/exponent to make sure it is not
+     * too short or long. Some impls have their own min and max key sizes that
+     * may or may not match with a system defined value.
+     *
+     * @param modulusLen
+     *            the bit length of the RSA modulus.
+     * @param exponent
+     *            the RSA exponent
+     * @param minModulusLen
+     *            if > 0, check to see if modulusLen is at least this long,
+     *            otherwise unused.
+     * @param maxModulusLen
+     *            caller will allow this max number of bits. Allow the smaller
+     *            of the system-defined maximum and this param.
+     * @param specificModulesLen
+     *            specific module length for sign/verify.
+     *
+     * @throws InvalidKeyException
+     *             if any of the values are unacceptable.
+     */
+    static void checkKeyLengths(int modulusLen, BigInteger exponent, int minModulusLen,
+            int maxModulusLen, int[] specificModulesLen) throws InvalidKeyException {
+
```

Comment:
Can this code make use of line 103 above to avoid duplicate code? Perhaps it checks all the parameters except for specificModulesLen using the method above then checks the values within `specificModulesLen`

Reply:
I updated the function by adding additional parameters, i think it'll be clear to use another overload function for sign and verify.

Reply:
It does still seem like there is duplicate code between this method and the other `checkKeyLengths` id suggest this new method implemented could call the other to check the values for `int modulusLen, BigInteger exponent, int minModulusLen,
int maxModulusLen` then add an additional check in this method for the specific modulus length check you are performing?

Reply:
updated